### DISCUSSION TITLE
ENOPKG Missing `package.json` file?

### DISCUSSION THREAD

**Discussion by jziggas** [semantic-release/semantic-release#3533] [Configuration]
https://github.com/swrpg-online/dice

This is my first npm package that I am trying to release and also my first time using semantic-release. I am finding the documentation extremely confusing with no clear step-by-step guidelines on how to get all of the configuration set up correctly from start to end.

The release step of my GH action file is currently failing at `ENOPKG Missing `package.json` file` and I can't figure out where te specific problem is now after spending several hours. Several different configurations between tsconfig, package.json, relase.yml, and .releaserc.json have been tried and now I no longer know up from down. 

I would think this package should look like any other small npm package with the one 
exception of also including a `cli.js` to be made available if the user installs the package globally.

Can anyone could point me in the right direction? I am willing to donate to this project to get the problem solved at this point because it is so frustrating. Thanks.

**Comment by jziggas** [comment]
Well.. it works now and unfortunately I have no idea what might have been the cause after so many iterations and getting GPT involved.

The last big change was the `.releaserc.json` from
```
{
  "branches": ["main"],
  "plugins": [
    "@semantic-release/commit-analyzer",
    "@semantic-release/release-notes-generator",
    "@semantic-release/changelog",
    [
      "@semantic-release/npm",
      {
        "npmPublish": true,
        "pkgRoot": "dist"
      }
    ],
    [
      "@semantic-release/github",
      {
        "assets": [{ "path": "dist/*.tgz", "label": "npm package" }]
      }
    ],
    "@semantic-release/git"
  ]
}
```

to a simpler:
```
{
  "branches": ["main"],
  "plugins": [
    "@semantic-release/commit-analyzer",
    "@semantic-release/release-notes-generator",
    "@semantic-release/changelog",
    "@semantic-release/npm",
    "@semantic-release/github"
  ]
}

```

maybe it had to do with the `pkgRoot` property.

**Comment by travi** [comment]
out of curiosity, why did you decide to add the git and changelog plugins? our docs recommend against them ([here](https://semantic-release.gitbook.io/semantic-release/support/faq#why-is-the-package.jsons-version-not-updated-in-my-repository) and [here](https://semantic-release.gitbook.io/semantic-release/support/faq#should-release-notes-be-committed-to-a-changelog.md-in-my-repository-during-a-release) as well as in the readmes of each of them). without adding those plugins, you could have used the default configuration and had no need for any releaserc at all
